### PR TITLE
ci: Add Tests For Ruby 3.3

### PR DESCRIPTION
**Pull request by arielvalentin**
Ruby 3.3 is out so this PR adds actions to test against it. 

One this to note is that `bigcommerce/gruf` does not yet support Ruby 3.3. Once https://github.com/bigcommerce/gruf/pull/197 is merged I will update the test suite to include gruf again.

### CODE REVIEW

File: .github/workflows/ci-instrumentation-with-services.yml [hunk 1]
```diff
@@ -69,6 +74,12 @@ jobs:
     runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@v4
+      - name: "Test Ruby 3.3"
+        uses: ./.github/actions/test_gem
+        with:
+          gem: "opentelemetry-${{ matrix.gem }}"
+          ruby: "3.3"
+          latest: "true"
```
Comment (arielvalentin) [inline .github/workflows/ci-instrumentation-with-services.yml:82]:
```suggestion
```

File: .github/workflows/ci-instrumentation-with-services.yml [hunk 2]
```diff
@@ -106,6 +117,12 @@ jobs:
...
```
Comment (arielvalentin) [inline .github/workflows/ci-instrumentation-with-services.yml:125]:
```suggestion
```

File: .github/workflows/ci-instrumentation-with-services.yml [hunk 3]
```diff
@@ -159,6 +176,12 @@ jobs:
...
```
Comment (arielvalentin) [inline .github/workflows/ci-instrumentation-with-services.yml:184]:
```suggestion
```

File: .github/workflows/ci-instrumentation-with-services.yml [hunk 4]
```diff
@@ -203,6 +226,12 @@ jobs:
...
```
Comment (arielvalentin) [inline .github/workflows/ci-instrumentation-with-services.yml:234]:
```suggestion
```

File: .github/workflows/ci-instrumentation-with-services.yml [hunk 5]
```diff
@@ -244,6 +273,12 @@ jobs:
...
```
Comment (arielvalentin) [inline .github/workflows/ci-instrumentation-with-services.yml:281]:
```suggestion
```

**Review by kreopelle — APPROVED** [review]

**Comment by arielvalentin** [comment]
@kreopelle did you mean to approve as your alter ego @kaylareopelle ?

**Review by plantfansam — APPROVED** [review]
🙏

File: .github/workflows/ci-instrumentation-canary.yml [hunk 6]
```diff
@@ -59,6 +59,14 @@ jobs:
     runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@v4
+      - name: "Test Ruby 3.3"
+        # BLOCKED BY: https://github.com/bigcommerce/gruf/pull/197
```
Comment (splittingred) [inline .github/workflows/ci-instrumentation-canary.yml:63]: Hey @arielvalentin - this has now been addressed and released, in connection with the grpc 1.61 release: https://rubygems.org/gems/gruf/versions/2.19.0

**Pull request merged**